### ISSUE TITLE
[bug] naming an argument `cmd` causes unexpected and undocumented error

### ISSUE THREAD

**Issue by stevepryde** [tauri-apps/tauri#4875]
### Describe the bug

I named an argument for a custom command `cmd`, which used to work previously but now it seems to be a special name that gets treated differently by tauri. In fact it seems to override the command specified as the first parameter to `invoke()`.

```rust
#[tauri::command]
pub fn my_command(
  cmd: String
) -> Result<String, String> {
    info!("This never gets logged");
    Ok(String::new())
}
```
I invoked it from JS like this:
```
window.__TAURI__.invoke("my_command", {cmd: "test"})
```
and instead of calling my_command and passing that argument, it returns the error message `command "test" not found` in the frontend and the backend command was never executed.

This seems really strange to me. Why would certain argument names be considered special?
Is this documented anywhere?

Given that there are such special argument names, could these please be documented here? or at least somewhere prominent?
https://tauri.app/v1/guides/features/command

Even better, I'd argue that any "special" args internal to tauri should be an optional 3rd parameter to invoke() rather than overloading the user-supplied args list.

### Reproduction

```rust
#[tauri::command]
pub fn my_command(
  cmd: String
) -> Result<String, String> {
    info!("This never gets logged");
    Ok(String::new())
}
```

Now try to invoke the above command. This doesn't work:
```
window.__TAURI__.invoke("my_command", {cmd: "test"})
```
Instead it gives the error `Uncaught command test not found`

### Expected behavior

I'd expect all of the arguments given in the 2nd parameter to `invoke()` to be delivered to my function regardless, and not to have any "magic" args that are treated differently. Any such special internal args should be in a separate parameter given to `invoke()`

### Platform and versions

Environment
  › OS: Windows 10.0.22000 X64
  › Webview2: 103.0.1264.77
  › MSVC:
  › Node.js: 16.13.0
  › npm: 8.1.0
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.25.1
  › rustc: 1.62.0
  › cargo: 1.62.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc

**Comment by itfanr** [comment]
Is it not fix in version   ` "@tauri-apps/api": "^1.1.0"` ? I encountered this bug when call a functhion like this:

```

#[tauri::command]
fn mycmd(cmd: &str) -> String{
    format!("Hello, {}! You've been greeted from Rust!", cmd)
}
```

**Comment by FabianLars** [comment]
> Is it not fix in version `"@tauri-apps/api": "^1.1.0"` ?

No, we can't fix this in a 1.X release because if people don't update the js _and_ rust packages _at the same time_, every js<->rust communication would break so we have to wait for v2 to fix it :/

**Comment by itfanr** [comment]
@FabianLars  Thank you for your reply.
